### PR TITLE
Improved Unit Testing - First Cut

### PR DESCRIPTION
**Pull request by pkanher617**
The following PR adds additional Unit Tests, as well as fixes or notes for bugs found during the first cut of testing.
Major Changes:
- Tests for merged changes now do not overwrite private functions or give false positives
- isInIframe made public
- Notes for future bug fixes

### CODE REVIEW

File: lib/msal-core/karma.conf.js [hunk 1]
```diff
@@ -7,16 +7,23 @@ module.exports = function (config) {
         frameworks: ['mocha', 'chai'],
         files: [
             'node_modules/babel-polyfill/dist/polyfill.js',
-            'tests/mochaTests/**/*.ts',
+            'tests/**/*.ts',
+        ],
+        exclude: [
+            'tests/MSAL*',
```
Comment (DarylThayil) [inline lib/msal-core/karma.conf.js]: should this have a file extension?

File: lib/msal-core/karma.conf.js [hunk 2]
```diff
@@ -7,16 +7,23 @@ module.exports = function (config) {
...
+            'tests/Authority.spec.ts'
```
Comment (DarylThayil) [inline lib/msal-core/karma.conf.js]: what happened to tests in mocha tests?

Comment (pkanher617) [reply lib/msal-core/karma.conf.js]: Moved into the UAA.spec file.

File: lib/msal-core/karma.conf.js (hunk 1 above)
Comment (pkanher617) [reply lib/msal-core/karma.conf.js]: this is just temporary, this will be gone in the future.

**Pull request merged**